pep-9999.rst

```diff
@@ -0,0 +1,520 @@
+
+In even-numbered years, the focus for both events would just be on the upcoming
+maintenance and beta releases. These less intense years would hopefully provide
+an opportunity to tackle various process changes and infrastructure upgrades
+without impacting the release candidate preparation process.
+
+
+Release cycle alignment for prominent Linux distributions
+---------------------------------------------------------
+
+Some rolling release Linux distributions (e.g. Arch, Gentoo) may be in a
+position to consume the new rolling beta releases proposed in this PEP, but it
+is expected that most distributions would continue to use the established
+releases.
+
+The specific dates for those releases proposed in this PEP are chosen to align
+with the feature freeze schedules for the annual October releases of the Ubuntu
+and Fedora Linux distributions.
+
+For both Fedora and Ubuntu, it means that the release candidate phase aligns
+with the development period for a distro release, which is the ideal time for
+them to test a new version and provide feedback on potential regressions and
+compatibility concerns.
+
+For Ubuntu, this also means that their April LTS releases will have benefited
+from a full short-term release cycle using the new system Python version, while
+still having that CPython release be open to upstream bug fixes for most of the
+time until the next Ubuntu LTS release.
+
+The one Linux release cycle alignment that is likely to be consistently poor
+with the specific proposal in this PEP is with Debian, as that has been released
+in the first half of odd-numbered years since 2005 (roughly 12 months offset
+from Ubuntu LTS releases).
+
+With the annual release proposal in PEP 602, both Debian and Ubuntu LTS would
+consistently get a system Python version that is around 6 months old, but
+would also consistently select different Python versions from each other.
+
+With a two year cadence, and CPython releases in the latter half of the year,
+they're likely to select the same version as each other, but one of them will
+be choosing a CPython release that is more than 18 months behind the latest beta
+releases by the time the Linux distribution ships.
+
+If that situation does occur, and is deemed undesirable (but not sufficiently
+undesirable for *Debian* to choose to adjust their release timing), then that's
+where the additional complexity of the "incremental feature release" proposal
+in PEP 598 may prove worthwhile.
+
+(Moving CPython releases to the same half of the year as the Debian and Ubuntu
+LTS releases would potentially help mitigate the problem, but also creates
+new problems where a slip in the CPython release schedule could directly affect
+the release schedule for a Linux distribution)
+
+
+
+Implications for simple deployment environments
+-----------------------------------------------
+
+TODO ()
+
+
+Implications for complex deployment environments
+------------------------------------------------
+
+TODO ()
```

Comment:
May I suggest: "assuming an organization with a centrally managed distribution"
* predictable CPython schedule means work can be scheduled
* probably has own package builds
* may have own patches, so being closer to the CPython beta schedule may make upstream contributions easier

Reply:
Also, I think the Python distros such as conda-forge, WinPython, Anaconda, etc. would fit into this category. It's probably worth reaching out to them for comment, as well as the Scientific Python section.

But perhaps we should get this first pass merged and start the discuss thread, then they can weigh in there.

Reply:
Probably needs some rewording, but here's my proposed text:

> For the purposes of this PEP, "complex" deployment environments are use cases which don't meet the "simple deployment" criterion above. They may involve multiple distinct versions of Python, use of a personalised build of Python, or "gatekeepers" who are required to approve use of a new version prior to deployment.
>
> For example, organisations that install Python on their users' machines as part of a standard operating environment fall into this category, as do those that provide a standard build environment. Distributions such as conda-forge or WinPython that provide collections of consistently built and verified packages are impacted in similar ways.
>
> These organisations tend to either prefer high stability (for example, all of those who are happily using Python 2.7) or fast turnaround (for example, those who regularly contribute toward the latest prereleases). In some cases, both may exist within the same organisation for different purposes (for example, a stable Python in the build environment and the most recent update for data scientists).
>
> Under any release model, each new release of Python generates work for these organisations. This work may involve legal, security or technical reviews of Python itself, assessment and verification of impactful changes, reapplication of patches, recompilation and testing of third-party dependencies, and deployment.
>
> Organisations that can take updates quickly should be able to make use of the more frequent beta releases. Each update will require similar investigative work to today, though as each release will be more similar to the previous than under the present model, the actual work required will be reduced.
>
> For organisations with stricter evaluations or a preference for stability, the longer release cycle for stable releases will reduce the annual effort required to update, the longer release candidate period will allow more time to do internal testing before the X.Y.0 release, and the greater use by others during the beta period will provide more confidence in the initial releases. Meanwhile, the organisation can confidently upgrade through maintenance releases for a longer time without fear of breaking changes.

Reply:
Suggestion regarding the third paragraph:

> These organisations tend to either prefer high stability (for example, all of those who are happily using Python 2.7) or fast turnaround (for example, those who regularly contribute toward the latest prereleases). In some cases, both may exist within the same organisation for different purposes (for example, a stable Python in the build environment and the most recent update for data scientists).

I think this might have too many occurrences of "for example". Also, given the imminent sunset of Python 2.7, we may want to avoid implicitly encouraging it's usage for those who prefer high stability.

Reply:
I have no problem with dropping some of the repeated phrases, but mentioning Python 2.7 here is very deliberate - the stability is _absolutely_ one of the reasons some people are happier there than on Python 3, and acknowledging those users is important, as is proposing a release process that will continue to support them.

Reply:
> mentioning Python 2.7 here is very deliberate - the stability is absolutely one of the reasons some people are happier there than on Python 3, and acknowledging those users is important

I can definitely agree that acknowledging those users is highly important, but would it be possible to do so without specifically referencing 2.7, such as "all of those who are happily using older versions"? The current version might not age especially well if someone were to read this PEP in a year or two from now, when Python 2.7 is well past the sunset and the final version was already released by Benjamin.